src/neptune_scale/sync/sync_process.py

```diff
@@ -131,7 +132,7 @@ class StatusTrackingElement(NamedTuple):
 def code_to_exception(code: IngestCode.ValueType) -> Optional[type[Exception]]:
     if code in CODE_TO_ERROR:
         return CODE_TO_ERROR[code]
-    return NeptuneUnexpectedError
+    return functools.partial(NeptuneUnexpectedError, reason=f"Unexpected error, code: {code}")
```

Comment:
This is a random drive-by fix: NeptuneUnexpectedError takes a mandatory parameter. Previously this codepath would crash when trying to instantiate it, hiding any information about original error and potentially messing up anyone trying to catch neptune errors (as opposed to random TypeErrors complaining about number of parameters in __init__)

Reply:
Doesn't matter, the same issue was fixed earlier today :)